From 974c17d2b6f570cd6325b2116e8051b2377c8c21 Mon Sep 17 00:00:00 2001
From: Ivan <spnethw@gmail.com>
Date: Thu, 2 Jan 2025 12:41:04 +0400
Subject: [PATCH] fix #2605: files highlighting and sort groups bugs

---
 far2l/src/hilight.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/far2l/src/hilight.cpp b/far2l/src/hilight.cpp
index 45b747a14..957a5f778 100644
--- a/far2l/src/hilight.cpp
+++ b/far2l/src/hilight.cpp
@@ -778,6 +778,7 @@ void HighlightFiles::HiEdit(int MenuPos)
 
 					{
 						ConfigWriter(RegColorsHighlight).RemoveSection();
+						ConfigWriter(SortGroupsKeyName).RemoveSection(); // see https://github.com/elfmz/far2l/issues/2605
 					}
 					HiMenu.Hide();
 					ClearData();
@@ -1032,6 +1033,13 @@ static void SaveFilter(FileFilterParams *CurHiData, ConfigWriter &cfg_writer, bo
 
 void HighlightFiles::SaveHiData()
 {
+	{
+		// hotfix for https://github.com/elfmz/far2l/issues/2605
+		// TODO: verify it's correct fix, not just a workaround
+		ConfigWriter(RegColorsHighlight).RemoveSection();
+		ConfigWriter(SortGroupsKeyName).RemoveSection();
+	}
+
 	std::string strRegKey, strGroupName;
 	const char *KeyNames[4] = {RegColorsHighlight, SortGroupsKeyName, SortGroupsKeyName, RegColorsHighlight};
 	const char *GroupNames[4] = {fmtFirstGroup, fmtUpperGroup, fmtLowerGroup, fmtLastGroup};
